sys/include/shell_lock.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2020 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup     sys_shell_lock Shell lock
+ * @ingroup      sys
+ * @brief        Simple module to provide a password protection for the shell.
+ * @experimental This module is an experimental feature and only shows as a PoC
+ *               how the shell could be protected with a password. Don't expect
+ *               relevant security from it for production, as Man-in-the-Middle
+ *               attacks are possible depending on the used connection method!
+ *
+ * @{
+ *
+ * @file
+ * @brief       Shell interface definition
+ */
+
+#ifndef SHELL_LOCK_H
+#define SHELL_LOCK_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "shell.h"
+
+/**
+ * @brief Lock the login process after given attempts of failed logins for
+ * a few seconds
+ */
+#define ATTEMPTS_BEFORE_TIME_LOCK 3
+
+#ifdef MODULE_SHELL_LOCK
+    #ifndef SHELL_LOCK_PASSWORD
+        #error Using MODULE_SHELL_LOCK requires defining SHELL_LOCK_PASSWORD
+    #endif /* SHELL_LOCK_PASSWORD */
+#endif /* MODULE_SHELL_LOCK */
+
+#ifdef MODULE_SHELL_LOCK_AUTO_LOCKING
+    /**
+     * @brief Lock the shell after this time span without user input
+     *        Defaults to 5 minutes and can be overwritten by defining
+     *        SHELL_LOCK_AUTO_LOCK_TIMEOUT_MS in the applications Makefile
+     */
+    #define MAX_AUTO_LOCK_PAUSE_MS 5 * 60 * 1000
+
+    #ifdef SHELL_LOCK_AUTO_LOCK_TIMEOUT_MS
+        #undef MAX_AUTO_LOCK_PAUSE_MS
+        #define MAX_AUTO_LOCK_PAUSE_MS SHELL_LOCK_AUTO_LOCK_TIMEOUT_MS
+    #endif /* SHELL_LOCK_AUTO_LOCK_TIMEOUT_MS */
+
+    /**
+     * @brief Offset used for the thread for automated locking, so that the
+     *        thread is not woken up shortely before it has to lock the shell.
+     */
+    #define TIMER_SLEEP_OFFSET_MS 100
```

Comment:
I don't understand why this is needed.

Reply:
A relic, the mentioned thread does not exist anymore.